common/src/reserved.rs

```diff
@@ -338,4 +462,189 @@ mod tests {
+            delegatee_public_key.clone(),
+            true,
+            0u64,
+        );
+        let proof = TypedSignature::sign(&data, &delegator_private_key).unwrap();
+
+        let tx = TxDelegate {
+            delegator: delegator_public_key.clone(),
+            delegatee: delegatee_public_key.clone(),
+            governance: true,
+            proof,
+            timestamp: 0,
+        };
+        let new_state = state.apply_delegate(&tx);
+
+        // then
+        assert!(
+            new_state.is_ok(),
+            "apply_delegate failed with {:?}",
+            new_state
+        );
+        let new_state = new_state.unwrap();
+
+        assert_eq!(
+            new_state.members[0]
+                .governance_delegations
+                .as_ref()
+                .unwrap(),
+            &delegatee.name.to_string()
+        );
+        assert_eq!(new_state.members[0].consensus_voting_power, 0);
+        assert_eq!(new_state.members[0].governance_voting_power, 0);
+        assert_eq!(
+            new_state.members[0].consensus_delegations,
+            Some(delegatee.name.to_string())
+        );
+        assert_eq!(
+            new_state.members[0].governance_delegations,
+            Some(delegatee.name.to_string())
+        );
+        assert_eq!(new_state.members[1].consensus_voting_power, 30);
+        assert_eq!(new_state.members[1].governance_voting_power, 30);
+    }
+
+    #[test]
+    fn test_apply_delegate_without_governance_success() {
+        // given
+        setup_test();
+        let (
+            delegator_public_key,
+            delegator_private_key,
+            delegatee_public_key,
+            delegatee,
+            mut state,
+        ) = setup_TxDelegate_test();
+
+        // when
+        let data = (
+            delegator_public_key.clone(),
+            delegatee_public_key.clone(),
+            true,
+            0u64,
+        );
+        let proof = TypedSignature::sign(&data, &delegator_private_key).unwrap();
+
+        let tx = TxDelegate {
+            delegator: delegator_public_key.clone(),
+            delegatee: delegatee_public_key.clone(),
+            governance: false,
+            proof,
+            timestamp: 0,
+        };
+        let new_state = state.apply_delegate(&tx);
+
+        // then
+        assert!(
+            new_state.is_ok(),
+            "apply_delegate failed with {:?}",
+            new_state
+        );
+        let new_state = new_state.unwrap();
+
+        assert_eq!(
+            new_state.members[0].consensus_delegations.as_ref().unwrap(),
+            &delegatee.name.to_string()
+        );
+        assert_eq!(new_state.members[0].consensus_voting_power, 0);
+        assert_eq!(new_state.members[0].governance_voting_power, 10);
+        assert_eq!(
+            new_state.members[0].consensus_delegations,
+            Some(delegatee.name.to_string())
+        );
+        assert_eq!(new_state.members[0].governance_delegations, None);
```

Comment:
The difference between the `with governance` and `without governance` test cases are the following.
* verifying whether the `governance_delegations` is None (without governance) or exists in the `MemberName` type (with governance)

This fact raises the following questions: should those who are delegated voting powers can be mapped in a one-to-many relationship? In other words, more than two validators can delegate each voting power to a single validator? If it so, the data type of Member's `governance_delegations` should be changed for the sake of including more than delegators.

Reply:
If a one-to-many relationship should be valid, I will make the following changes:
```Rust
#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
pub struct Member {
    pub public_key: PublicKey,
    ...
    pub governance_delegators: Option<Members> ...
}

pub fn get_validator_set(&self) -> Result<Vec<(PublicKey, VotingPower)>, String> {
        let mut validator_set = HashMap::new()
        ...
                validator_set
                    .entry(member.name.clone())
                    .and_modify(|v| *v += member.consensus_voting_power)
                    .and_modify(|v| *v.update_governance_delegators(delegator) // like this
                    .or_insert(member.consensus_voting_power);
            }
        }
    }
```

Reply:
Additionally, the `nit` point should be considered to separate `get_validator_set` and `update_validator_set` for the sake of isolation of read/update features.